controller/src/statemachine.rs

```diff
@@ -36,14 +83,27 @@ pub fn determine_next_node_spec(brs: &BottlerocketShadow) -> BottlerocketShadowS
                     BottlerocketShadowSpec::new_starting_now(
                         brs.spec.state.on_success(),
                         brs.spec.version(),
+                        0,
+                        None,
                     )
                 }
                 // In any other circumstance, we just proceed to the next step.
                 _ => BottlerocketShadowSpec::new_starting_now(
                     brs.spec.state.on_success(),
                     brs.spec.version(),
+                    brs.spec.crash_count(),
+                    brs.spec.failure_timestamp().unwrap(),
                 ),
             }
         }
     }
 }
+
+fn calculated_time_since_crash_in_minutes(brs: &BottlerocketShadow) -> Option<i64> {
```

Comment:
The reason that I use the `Option` here was I want to simplify the logic to check crash/not crash. I feel return i64 rather than Duration would make it caller's logic easier. Otherwise the caller would need to compare `crash_time_period> Duration::minutes(2_i64.pow(brs.spec.crash_count().try_into().unwrap()))`. 

I will decide which to use with the implementation for exponential backoff  as a separate function.

Reply:
Yeah. I think abstracting the comparison code into its own function might be preferred, we should try to use the most appropriate type here to represent the return value.